examples/step-78/step-78.cc

```diff
@@ -0,0 +1,889 @@
+    preconditioner.initialize(system_matrix, 1.0);
+    cg.solve(system_matrix, solution, system_rhs, preconditioner);
+    constraints.distribute(solution);
+    std::cout << "     " << solver_control.last_step() << " CG iterations."
+              << std::endl;
+  }
+
+  // @sect4{<code>BlackScholes::add_results_for_output</code>}
+
+  // This is simply the function to build the solution together. For this, we
+  // create a new layer at each time, and then add the solution vector for that
+  // timestep. The function then stitches this together with the old solutions
+  // using 'build_patches'.
+  template <int dim>
+  void BlackScholes<dim>::add_results_for_output()
+  {
+    data_out_stack.new_parameter_value(time, time_step);
+    data_out_stack.attach_dof_handler(dof_handler);
+    data_out_stack.add_data_vector(solution, solution_names);
+    data_out_stack.build_patches(2);
+    data_out_stack.finish_parameter_value();
+  }
+
+  // @sect4{<code>BlackScholes::refine_grid</code>}
+
+  // This is somewhat unnecessary to have a function for the global refinement
+  // that we do. The reason for the function, is to allow for the possibility of
+  // an adaptive refinement later.
+  template <int dim>
+  void BlackScholes<dim>::refine_grid()
+  {
+    triangulation.refine_global(1);
+  }
+
+  // @sect4{<code>BlackScholes::process_solution</code>}
+
+  // This is where we calculate the convergence and error data to evaluate the
+  // effectiveness of the program. Here, we calculate the $L^2$, $H^1$ and
+  // $L^{\infty}$ norms.
+  template <int dim>
+  void BlackScholes<dim>::process_solution()
+  {
+    Solution<dim> sol(maturity_time);
+    sol.set_time(time);
+    Vector<float> difference_per_cell(triangulation.n_active_cells());
+    VectorTools::integrate_difference(dof_handler,
+                                      solution,
+                                      sol,
+                                      difference_per_cell,
+                                      QGauss<dim>(fe.degree + 1),
+                                      VectorTools::L2_norm);
+    const double L2_error =
+      VectorTools::compute_global_error(triangulation,
+                                        difference_per_cell,
+                                        VectorTools::L2_norm);
+    VectorTools::integrate_difference(dof_handler,
+                                      solution,
+                                      sol,
+                                      difference_per_cell,
+                                      QGauss<dim>(fe.degree + 1),
+                                      VectorTools::H1_seminorm);
+    const double H1_error =
+      VectorTools::compute_global_error(triangulation,
+                                        difference_per_cell,
+                                        VectorTools::H1_seminorm);
+    const QTrapez<1>  q_trapez;
+    const QIterated<dim> q_iterated(q_trapez, fe.degree * 2 + 1);
```

Comment:
```suggestion
    const QTrapezoid<1>  q_trapezoid;
    const QIterated<dim> q_iterated(q_trapezoid, fe.degree * 2 + 1);
```
`QTrapez` has been flagged deprecated. That's the reason why the CI runners complain:
```
/home/runner/work/dealii/dealii/examples/step-78/step-78.cc: In member function ‘void BlackScholesSolver::BlackScholes<dim>::process_solution()’:
/home/runner/work/dealii/dealii/examples/step-78/step-78.cc:578:23: error: ‘using QTrapez = class dealii::QTrapezoid<1>’ is deprecated [-Werror=deprecated-declarations]
     const QTrapez<1>  q_trapez;
                       ^~~~~~~~
In file included from /home/runner/work/dealii/dealii/examples/step-78/step-78.cc:28:0:
/home/runner/work/dealii/dealii/include/deal.II/base/quadrature_lib.h:137:51: note: declared here
 using QTrapez DEAL_II_DEPRECATED = QTrapezoid<dim>;
                                                   ^
```

Reply:
Sorry, I made this change temporarily when I was testing that everything still worked on my machine and I forgot to change it back...